controlplane/kubeadm/internal/workload_cluster_etcd.go

```diff
@@ -57,19 +63,24 @@ func (w *Workload) EtcdIsHealthy(ctx context.Context) (HealthCheckResult, error)
 			continue
 		}
 
-		// Check to see if the pod is ready
-		etcdPodKey := ctrlclient.ObjectKey{
-			Namespace: metav1.NamespaceSystem,
-			Name:      staticPodName("etcd", name),
-		}
-		pod := corev1.Pod{}
-		if err := w.Client.Get(ctx, etcdPodKey, &pod); err != nil {
-			response[name] = errors.Wrap(err, "failed to get etcd pod")
-			continue
+		for _, m := range controlPlane.Machines {
+			if m.Spec.ProviderID != nil && *m.Spec.ProviderID == node.Spec.ProviderID {
+				owningMachine = m
+				// Only set this condition if the node has an owning machine.
+				conditions.MarkTrue(owningMachine, clusterv1.MachineEtcdMemberHealthyCondition)
+				break
+			}
 		}
-		if err := checkStaticPodReadyCondition(pod); err != nil {
+
+		// TODO: If owning machine is nil, should not continue. But this change breaks the logic below.
+
+		// Check etcd pod's health
+		isEtcdPodHealthy, _ := w.checkPodStatusAndUpdateCondition(EtcdPodNamePrefix, node, owningMachine, clusterv1.MachineEtcdPodHealthyCondition)
+		// isEtcdPodHealthy is false, if Pod is not Ready or there is a client error.
+		if !isEtcdPodHealthy {
```

Comment:
Why are we ignoring the error her?

Reply:
It was only checking if etcd Pod is Ready or not before. I don't want to change that behaviour in this PR.